app/pages/taxpayer/TaxpayerLoopPage.scala

```diff
@@ -28,7 +28,7 @@ case object TaxpayerLoopPage extends LoopPage[Taxpayer] {
   override def toString: String = "taxpayerLoop"
 
   def updatedLoopList(userAnswers: UserAnswers, id: Int): IndexedSeq[Taxpayer] = {
-    val buildTaxpayer: Taxpayer = Taxpayer.buildTaxpayerDetails(userAnswers, id)
+    val buildTaxpayer: Taxpayer = Taxpayer(userAnswers, id)
     userAnswers.get(TaxpayerLoopPage, id) match {
       case Some(list) => // append to existing list without duplication
         list :+ buildTaxpayer
```

Comment:
We should avoid taxpayer duplication by id:

`list.filterNot(_.taxpayerId == buildTaxpayer.taxpayerId) :+ buildTaxpayer `